python-lib/dku_deeplearning_image/utils.py

```diff
@@ -169,24 +175,50 @@ def format_predictions_output(predictions, classify=False, labels_df=None, limit
         return predictions.tolist()
     formatted_predictions = []
     id_pred = lambda index: labels_df.loc[index][constants.LABEL] if labels_df is not None else str(index)
+    pred_errors = []
     for pred in predictions:
-        formatted_pred = get_ordered_dict(
-            {id_pred(i): float(pred[i]) for i in pred.argsort()[-limit:] if float(pred[i]) >= min_threshold})
-        formatted_predictions.append(formatted_pred)
-    return formatted_predictions
-
-
-def get_predictions(model, batch, classify=False, limit=constants.DEFAULT_PRED_LIMIT, min_threshold=0, labels_df=None):
-    predictions = model.predict(batch)
-    return format_predictions_output(predictions, classify, labels_df, limit, min_threshold)
+        try:
+            formatted_pred = get_ordered_dict(
+                {id_pred(i): float(pred[i]) for i in pred.argsort()[-limit:] if float(pred[i]) >= min_threshold})
+            formatted_predictions.append(formatted_pred)
+            pred_errors.append(0)
+        except Exception:
+            pred_errors.append(1)
+    return {"prediction": formatted_predictions, "error": pred_errors}
 
 
 def get_ordered_dict(predictions):
     return json.dumps(OrderedDict(sorted(predictions.items(), key=(lambda x: -x[1]))))
 
 
-def preprocess_img(img_path, img_shape, preprocessing):
+def apply_preprocess_image(tfds, input_shape, preprocessing, is_b64=False):
+    def _apply_preprocess_image(image_path):
+        return tf.numpy_function(
+            func=lambda x: preprocess_img(x, input_shape, preprocessing, is_b64),
+            inp=[image_path],
+            Tout=tf.float32)
+
+    return tfds.map(map_func=_apply_preprocess_image, num_parallel_calls=constants.AUTOTUNE)
+
+
+def read_images_to_tfds(images_folder, np_images):
```

Comment:
Why split reading & preprocessing images ? They are both using custom code, and I think it would be clearer to merge them.

Reply:
Actually, I did that because when using a b64 image, the way of reading is different, I only need to preprocess the image. So I separated the two functions to make it possible.